pycheribuild/config/compilation_targets.py

```diff
@@ -211,17 +214,25 @@ def riscv_arch_string(self):
     def riscv_abi(self):
         assert self.target.is_riscv(include_purecap=True)
         if self.target.is_cheri_purecap():
-            return "l64pc128d"  # 64-bit double-precision hard-float + purecap
+            # 64-bit double-precision hard-float + purecap
+            return "l64pc128d" if self.target.cpu_architecture == CPUArchitecture.RISCV64 else \
+                   "il32pc64f"  # 32-bit single-precision soft-float + purecap
```

Comment:
Soft-float means no f.

Reply:
Shouldn't this include f? I.e. the comment is wrong not the ABI? Or do we really want to assume that softfloat for all rv32 targets?

Reply:
Well it needs to be soft-float if the arch string doesn't have F. In theory we could have an FD hard-float RV32 implementation but we don't currently and it's not something that will be used in SSITH.

Reply:
yeah, but this is the generic case. IMO there should just be a supports_hardfloat option that the newlib targetinfo sets to false and then the ABI and arch_string can match again.

Reply:
Well actually newlib supports it just fine (I did setjmp/longjmp properly), it's just that the FreeRTOS context switch code doesn't save/restore floating point registers.

Reply:
Yeah, also RTEMS/newlib RISC-V32 support hard-float. I fixed the comment and made RISC-V32 hard-float default to `il32pc64d` for ports that may (and choose to) support it (eg, RTEMS on spike/qemu)